matRad_calcDoseDirect.m

```diff
@@ -52,16 +52,20 @@
 end
 
 % dose calculation
-if strcmp(pln.radiationMode,'photons')
-    dij = matRad_calcPhotonDose(ct,stf,pln,cst,calcDoseDirect);
-    %dij = matRad_calcPhotonDoseVmc(ct,stf,pln,cst,5000,4,calcDoseDirect);
-elseif strcmp(pln.radiationMode,'protons') || strcmp(pln.radiationMode,'carbon')
-    dij = matRad_calcParticleDose(ct,stf,pln,cst,calcDoseDirect);
+dij = cell(length(stf),1);
+totalNumOfBixels = sum([stf(:).totalNumOfBixels]);
+for i = 1:length(stf)
+    if strcmp(pln.radiationMode,'photons')
+        dij{i} = matRad_calcPhotonDose(ct,stf(i),pln,cst,calcDoseDirect);
+        %dij = matRad_calcPhotonDoseVmc(ct,stf,pln,cst,5000,4,calcDoseDirect);
+    elseif strcmp(pln.radiationMode,'protons') || strcmp(pln.radiationMode,'carbon')
+        dij{i} = matRad_calcParticleDose(ct,stf(i),pln,cst,calcDoseDirect);
+    end
 end
 
 % remember bixel weight
 counter = 0;
-resultGUI.w = NaN * ones(dij.totalNumOfBixels,1);
+resultGUI.w = NaN * ones(dij{1}.totalNumOfBixels,1);
```

Comment:
Here you initialize with the wrong array size but you are saved as matlab automatically adjusts the vector!